### PR TITLE
feat!: use preferred formats for `MappableConcept.mappings`

### LINKED ISSUE #212: Use preferred formats for `MappableConcept.mappings`
### Feature description

Use the preferred `Coding` format for each item in `MappableConcept.mappings`. A `Coding` consists of a `relation` and `coding`, where a [Coding](https://github.com/ga4gh/gks-core/blob/1.0.0-snapshot.2024-11.2/schema/gks-core/json/Coding) requires: 
  - `code`: "A symbol uniquely identifying the concept, as in a syntax defined by the code system. CURIE format is preferred where possible (e.g. 'SO:0000704' is the CURIE form of the Sequence Ontology code for 'gene')."
  - `system`: description": "The terminology/code system that defined the code. May be reported as a free-text name (e.g. 'Sequence Ontology'), but it is preferable to provide a uri/url for the system. When the 'code' is reported as a CURIE, the 'system' should be reported as the uri that the CURIE's prefix expands to (e.g. 'http://purl.obofoundry.org/so.owl/' for the Sequence Ontology)."

### Use case

The `Coding` object allows for flexibility in how `system` and `code` are represented, which can lead to inconsistency across different implementations and systems. Aligning these fields to use preferred formats can help ensure consistent representation.

### Acceptance Criteria

Given a Mappable Concept (disease) is queried,
When it has `mappings`,
Then the `system` must be represented as a URI/URL and the `code` must be represented as a CURIE, where possible

### Proposed solution

_No response_

### Alternatives considered

_No response_

### Implementation details

_No response_

### Potential Impact

_No response_

### Additional context

_No response_

### Contribution

Yes, I can create a PR for this feature.

### PR DESCRIPTION
**Pull request by korikuzma**
close #212

* `system` will use OBO Foundry persistent URL (PURL), source homepage, or namespace prefix, in that order of preference, if available.
* `code` will use the `concept_id` as the CURIE

### CODE REVIEW

**Comment by korikuzma** [comment]
Ope, forgot to update schemas / check docs. One sec

**Review by jsstevenson — APPROVED** [review]

File: src/disease/schemas.py [hunk 1]
```diff
@@ -88,6 +88,36 @@ class NamespacePrefix(Enum):
     WIKIDATA = "wikidata"
 
 
+# Source to URI (from identifiers.org, if found)
+NAMESPACE_TO_SYSTEM_URI: dict[NamespacePrefix, str] = {
```
Comment (jsstevenson) [inline src/disease/schemas.py:92]: maybe this is my own misunderstanding, but shouldn't we be using pURLs where possible here?

Comment (korikuzma) [reply src/disease/schemas.py:92]: I think this is part of the ambiguity with this. I can update

File: src/disease/schemas.py [hunk 2]
```diff
@@ -88,6 +88,36 @@ class NamespacePrefix(Enum):
...
+    NamespacePrefix.MEDDRA: "https://purl.bioontology.org/ontology/MEDDRA/",
+    NamespacePrefix.MEDGEN: "https://www.ncbi.nlm.nih.gov/medgen/",
+    NamespacePrefix.MESH: "https://id.nlm.nih.gov/mesh/",
+    NamespacePrefix.MP: "https://www.ebi.ac.uk/ols4/ontologies/mp/terms?obo_id=MP:",
+    NamespacePrefix.OBI: "https://purl.obolibrary.org/obo/",
+    NamespacePrefix.ORPHANET: "https://www.orpha.net/consor/cgi-bin/OC_Exp.php?Lng=EN&Expert=",
+    NamespacePrefix.PATO: "https://www.ebi.ac.uk/ols4/ontologies/pato/terms?obo_id=PATO:",
+    NamespacePrefix.UMLS: "https://linkedlifedata.com/resource/umls/id/",
+    NamespacePrefix.WIKIPEDIA: "https://en.wikipedia.org/wiki/",
+    NamespacePrefix.WIKIDATA: "https://www.wikidata.org/entity/",
```
Comment (jsstevenson) [inline src/disease/schemas.py:112]: these would be my suggestions

* oncotree is pretty arbitrary but the below is what's provided by the bioregistry

```suggestion
    NamespacePrefix.NCIT: "http://purl.obolibrary.org/obo/NCIT_",
    NamespacePrefix.MONDO: "http://purl.obolibrary.org/obo/MONDO_",
    NamespacePrefix.DO: "http://purl.obolibrary.org/obo/DOID_",
    NamespacePrefix.DOID: "http://purl.obolibrary.org/obo/DOID_",
    NamespacePrefix.OMIM: "https://omim.org/MIM:",
    NamespacePrefix.ONCOTREE: "http://oncotree.mskcc.org/api/tumorTypes/search/code/",
    NamespacePrefix.EFO: "https://www.ebi.ac.uk/efo/EFO_",
    NamespacePrefix.HP: "http://purl.obolibrary.org/obo/HP_",
    NamespacePrefix.HPO: "http://purl.obolibrary.org/obo/HP_",
    NamespacePrefix.KEGG: "https://www.kegg.jp/entry/",
    NamespacePrefix.MEDDRA: "https://purl.bioontology.org/ontology/MEDDRA/",
    NamespacePrefix.MEDGEN: "https://www.ncbi.nlm.nih.gov/medgen/",
    NamespacePrefix.MESH: "https://id.nlm.nih.gov/mesh/",
    NamespacePrefix.MP: "http://purl.obolibrary.org/obo/MP_",
    NamespacePrefix.OBI: "http://purl.obolibrary.org/obo/OBI_",
    NamespacePrefix.ORPHANET: "https://www.orpha.net/consor/cgi-bin/OC_Exp.php?Lng=EN&Expert=",
    NamespacePrefix.PATO: "http://purl.obolibrary.org/obo/PATO_",
    NamespacePrefix.UMLS: "https://linkedlifedata.com/resource/umls/id/",
    NamespacePrefix.WIKIPEDIA: "https://en.wikipedia.org/wiki/",
    NamespacePrefix.WIKIDATA: "https://www.wikidata.org/entity/",
```

Comment (korikuzma) [reply src/disease/schemas.py:112]: I'm going to follow the description in [this](https://github.com/ga4gh/gks-core/pull/107) PR 

> The terminology/code system that defined the code. May be reported as a free-text name 
(e.g. 'Sequence Ontology'), but it is preferrable to provide an IRI or URL that resolves to 
          a specification of the system, or its home page. e.g. for the Sequence Ontology, 
          'http://purl.obolibrary.org/so.owl' resolves to an owl rendering of the ontology, and
          'http://www.sequenceontology.org/' resolves to its home page.

**Review by jsstevenson — APPROVED** [review]

**Review by jsstevenson — APPROVED** [review]

**Pull request merged**